wallet/wallet-rpc-lib/src/rpc/types.rs

```diff
@@ -428,11 +428,34 @@ impl StakingStatus {
         }
     }
 }
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, HasValueHint)]
+#[serde(tag = "source")]
```

Comment:
seems a bit weird that sometimes we use "source" sometimes "type" for the tag, should we agree on a single one?

Reply:
> seems a bit weird that sometimes we use "source" sometimes "type" for the tag, should we agree on a single one?

The idea was to use a name that describes what the enum discriminates over, falling back on "type" normally. I am open to suggestions / critiques.